include/CLI/TypeTools.hpp

```diff
@@ -74,6 +89,20 @@ template <typename T> struct is_copyable_ptr {
     static bool const value = is_shared_ptr<T>::value || std::is_pointer<T>::value;
 };
 
+template <typename T, typename Enable = void> struct is_path : std::false_type {};
+template <typename T, typename Enable = void> struct force_widen : std::false_type {};
+
+#if defined CLI11_HAS_FILESYSTEM && CLI11_HAS_FILESYSTEM > 0
+template <typename T>
+struct is_path<T, typename std::enable_if<std::is_same<std::filesystem::path, remove_cvref_t<T>>::value>::type>
+    : std::true_type {};
+#if _WIN32
+template <typename T> struct force_widen<T, enable_if_t<is_path<T>::value>> : std::true_type {};
+#endif
+#else
+template <typename T> struct is_path : std::false_type {};
+#endif
```

Comment:
Detect if T is a path, exists even if <filesystem> isn't used. I'm using remove_cvref, but that may not be necessary (but it doesn't hurt much)

if WIN32 => force_widen

Reply:
I wrote it this way so force_widen could be extended to other types later if needed.